backend/compact-connect/lambdas/python/search/handlers/provider_update_ingest.py

```diff
@@ -0,0 +1,217 @@
+Lambda handler to process SQS messages containing DynamoDB stream events and index
+provider documents into OpenSearch.
+
+This Lambda is triggered by SQS (via EventBridge Pipe from DynamoDB streams) from
+the provider table. It processes events in batches, deduplicates provider IDs by
+compact, and bulk indexes the sanitized provider documents into the appropriate
+OpenSearch indices.
+
+The handler uses the @sqs_batch_handler decorator which passes all SQS messages
+to the handler at once, enabling batch processing and deduplication. The handler
+returns batchItemFailures directly for partial success handling.
+"""
+
+from cc_common.config import config, logger
+from cc_common.exceptions import CCInternalException, CCNotFoundException
+from cc_common.utils import sqs_batch_handler
+from marshmallow import ValidationError
+from opensearch_client import OpenSearchClient
+from utils import generate_provider_opensearch_document
+
+
+@sqs_batch_handler
+def provider_update_ingest_handler(records: list[dict]) -> dict:
+    """
+    Process DynamoDB stream events from SQS and index provider documents into OpenSearch.
+
+    This function:
+    1. Creates a set for each compact to deduplicate provider IDs
+    2. Extracts compact and providerId from each stream record (old or new image)
+    3. Processes each unique provider by compact using the shared utility
+    4. Bulk indexes the documents into the appropriate OpenSearch index
+
+    :param records: List of SQS records, each containing 'messageId' and 'body' (DynamoDB stream record)
+    :return: Response with batch item failures for partial success handling
+    """
+    if not records:
+        logger.info('No records to process')
+        return {'batchItemFailures': []}
+
+    logger.info('Processing SQS batch with DynamoDB stream records', record_count=len(records))
+
+    # Create a set for each compact to deduplicate provider IDs
+    providers_by_compact: dict[str, set[str]] = {compact: set() for compact in config.compacts}
+
+    # Track which message IDs correspond to which compact/provider for failure reporting
+    record_mapping: dict[str, tuple[str, str]] = {}  # message_id -> (compact, provider_id)
+
+    # Extract compact and providerId from each record
+    for record in records:
+        message_id = record['messageId']
+        # The body contains the DynamoDB stream record sent via EventBridge Pipe
+        stream_record = record['body']
+
+        # Try to get the data from NewImage first, fall back to OldImage for deletes
+        image = stream_record.get('dynamodb', {}).get('NewImage') or stream_record.get('dynamodb', {}).get('OldImage')
+
+        if not image:
+            logger.error('Record has no image data', message_id=message_id, record=stream_record)
+            continue
+
+        # Extract compact and providerId from the DynamoDB image
+        # The format is {'S': 'value'} for string attributes
+        compact = _extract_string_value(image.get('compact'))
```

Comment:
You could just use the [TypeDeserializer](https://boto3.amazonaws.com/v1/documentation/api/latest/_modules/boto3/dynamodb/types.html) to deserialize the whole image and be done.